### PR TITLE
Remove os-specific IsTerminal methods

### LINKED ISSUE #599: undefined: windows.Syscall
I'm on

> ```
> go version go1.8 windows/amd64
> ```
> 
and even just downloading logrus causes:
> ```
> src\github.com\sirupsen\logrus\terminal_windows.go:78: undefined: windows.Syscall
> ```

I'm pretty sure my Go installation is not modified and according to https://godoc.org/golang.org/x/sys/windows there is in fact no function `Syscall()` in `golang.org/x/sys/windows`. I don't understand where it's supposed to come from.

### PR DESCRIPTION
**Pull request by dmathieu**
Use x/crypto/ssh/terminal instead

Closes #599 
Closes #603 
Closes #602 

Ping @sirupsen, there might be specificities I'm missing preventing us from doing this simplification.

### CODE REVIEW

**Comment by sirupsen** [comment]
@dmathieu interesting, when was this put into the standard library?

**Comment by dmathieu** [comment]
Looks like it's been around since [2012](https://github.com/golang/crypto/commit/521edf84a06b2769f7a673319dcaee4024d4ecff).

**Comment by dmathieu** [comment]
I'm going to merge this. The bug is causing too many issues anyway.

File: text_formatter.go [hunk 1]
```diff
@@ -61,7 +65,16 @@ type TextFormatter struct {
...
 func (f *TextFormatter) init(entry *Entry) {
 	if entry.Logger != nil {
-		f.isTerminal = IsTerminal(entry.Logger.Out)
+		f.isTerminal = f.checkIfTerminal(entry.Logger.Out)
+	}
+}
+
+func (f *TextFormatter) checkIfTerminal(w io.Writer) bool {
+	switch v := w.(type) {
+	case *os.File:
```
Comment (sirupsen) [inline text_formatter.go]: Sorry for the delay. Is `Stderr` considered an `os.File` by Go?

Comment (dmathieu) [inline text_formatter.go]: [it should be, yes](https://golang.org/pkg/os/#pkg-variables). But this hasn't changed. We're still checking the type the exact same way we were before.

Comment (sirupsen) [inline text_formatter.go]: Ah yes, I should've seen that. :-)

One note is that this actually breaks backwards compatibility. `IsTerminal` is unfortunately exported, but we recently changed the signature (didn't realize people relied on it) and it broke things for a few people. I think it's reasonable to _not_ provide this going forwards—just a heads up.

Comment (dmathieu) [inline text_formatter.go]: This is on master. If we do a new 1-x release, we shouldn't cherry-pick it.

**Comment by clnperez** [comment]
Soooo it looks like this re-broke ppc64le, since the crypto package is *still* using the now-deprecated syscall package. So close! ;)

**Comment by tsuna** [comment]
Sorry for the rant but pulling in over 1000 lines of code from `ssh/terminal` just for `isatty()` doesn't strike me as a great move... I understand it's annoying that this code varies so much across platforms and that there is nothing in the standard library or in `golang.org/x/sys/unix`, but come on...

**Pull request merged**